pkg/apiserver/watch.go

```diff
@@ -196,7 +197,9 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 				// client disconnect.
 				return
 			}
-			flusher.Flush()
+			if len(ch) == 0 {
```

Comment:
It wouldn't wait - once you reached the net/http buffer you'd get written.
I think the buffer is 32kb or 64kb by default.

On Mon, May 23, 2016 at 1:22 PM, Wojciech Tyczynski <
notifications@github.com> wrote:

> In pkg/apiserver/watch.go
> <https://github.com/kubernetes/kubernetes/pull/26085#discussion_r64254422>
> :
>
> > @@ -196,7 +197,9 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
> >  				// client disconnect.
> >  				return
> >  			}
> > -			flusher.Flush()
> > +			if len(ch) == 0 {
>
> Do we have any guarantee on how frequently it will happen?
> I know that this will happen, but if it waits for say tens of MB, that may
> not be good enough for us.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/26085/files/c4bec1585fe9c31303c928c11ac45e380d2eaf77#r64254422>
>